### PR TITLE
ci: add tidy lint

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong Dai <loong.dai@intel.com>

### CODE REVIEW

File: tools/make/golang.mk [hunk 1]
```diff
@@ -42,17 +44,10 @@ go.clean: ## Clean the building output files
...
-	@go install golang.org/x/tools/cmd/goimports
-endif
-
-.PHONY: go.format
-go.format:  go.format.verify
-	@echo "===========> Running go codes format"
-	@gofmt -s -w .
-	@goimports -w -local $(ROOT_PACKAGE) .
-	@go mod tidy
+.PHONY: go.tidy
```
Comment (arkodg) [inline tools/make/golang.mk:47]: you also might want to edit the top level `format` target and reuse `go.tidy` there
https://github.com/envoyproxy/gateway/blob/80b78e5521a0ce32274c36a845da3955df4aec09/Makefile.targets.mk#L76

Comment (daixiang0) [reply tools/make/golang.mk:49]: Done. Please review again.

File: tools/make/golang.mk [hunk 2]
```diff
@@ -73,9 +68,9 @@ build.multiarch: ## Build envoy-gateway for multiple platforms. See Option PLATF
 test: ## Run all Go test of code sources.
 	@$(MAKE) go.test.unit
 
-.PHONY: format
-format: ## Format codes style with mod tidy, gofmt and goimports.
-	@$(MAKE) go.format
+.PHONY: tidy
```
[the remark this answers is not in the document]
Comment (daixiang0) [reply tools/make/golang.mk]: Updated, thanks for review.

**Review by arkodg — APPROVED** [review]
🚀

**Pull request merged**